src/main/groovy/com/ullink/Msbuild.groovy

```diff
@@ -44,6 +44,8 @@ class Msbuild extends ConventionTask {
             project.file(project.name + ".csproj").exists() ? project.name + ".csproj" : null
         }
         conventionMapping.map "projectName", { project.name }
+        
+        outputs.files { projects.collect { _, project -> project.dotnetArtifacts.files }.flatten() }
```

Comment:
indeed it didn't work without having all the files referenced as input, I tested a little bit of your code but it's working for me, may be it's hard to suit all cases in msbuild

I was adding outputs.files hoping to have outputs.files set so it could be generalized and use in subsequent task input, but missed also the localization resources which is not included, I'll just remove it

but localization dll is not working neither in dotnetArtifacts though I couldn't find any useful output from ProjectFileParser